packages/tx/test/inputValue.spec.ts

```diff
@@ -0,0 +1,134 @@
+import tape from 'tape'
+import { Address, AddressLike, BN, BNLike, BufferLike, toBuffer } from 'ethereumjs-util'
+import Common, { Chain, Hardfork } from '@ethereumjs/common'
+import { Transaction } from '../src'
+
+// @returns: Array with subtypes of the AddressLike type for a given address
+function generateAddressLikeValues(address: string): AddressLike[] {
+  return [address, toBuffer(address), new Address(toBuffer(address))]
+}
+
+// @returns: Array with subtypes of the BNLike type for a given number
+function generateBNLikeValues(value: number): BNLike[] {
+  return [value, new BN(value), `0x${value.toString(16)}`, toBuffer(value)]
+}
+
+// @returns: Array with subtypes of the BufferLike type for a given string
+function generateBufferLikeValues(value: string): BufferLike[] {
+  return [value, toBuffer(value)]
+}
+
+interface GenerateCombinationsArgs {
+  options: { [x: string]: any }
+  optionIndex?: number
+  results?: { [x: string]: any }[]
+  current?: { [x: string]: any }
+}
+
+export function generateCombinations({
+  options,
+  optionIndex = 0,
+  results = [],
+  current = {},
+}: GenerateCombinationsArgs) {
+  const allKeys = Object.keys(options)
+  const optionKey = allKeys[optionIndex]
+
+  const values = options[optionKey]
+
+  for (let i = 0; i < values.length; i++) {
+    current[optionKey] = values[i]
+
+    if (optionIndex + 1 < allKeys.length) {
+      generateCombinations({ options, optionIndex: optionIndex + 1, results, current })
+    } else {
+      results.push(current)
+    }
+  }
+
+  return results
+}
+
+// Deterministic pseudorandom number generator
+function mulberry32(seed: number) {
+  let t = (seed += 0x6d2b79f5)
+  t = Math.imul(t ^ (t >>> 15), t | 1)
+  t ^= t + Math.imul(t ^ (t >>> 7), t | 61)
+  return ((t ^ (t >>> 14)) >>> 0) / 4294967296
+}
+
+function getRandomSubarray(array: any[], size: number) {
+  const shuffled = array.slice(0),
+    seed = 1559
+  let index: number,
+    i = array.length,
+    temp: any[]
+  while (i--) {
```

Comment:
Here again, more "I'm not a javascript guru" questions but presumably `while (i--)` is just shorthand for `while (i === 0) { i--} `?  I dropped some console logs in the test and that seems to be what's happening but just wanted to confirm.

Reply:
It's a shorthand for 
```
while (i > 0) {
 i = i - 1;
}
```

Reply:
oh nice didn't know this one. maybe you can also do one declaration per line for the above let variables as well. i wonder if there's a eslint rule to prefer one line per declaration 

found it: ^_^ https://eslint.org/docs/rules/one-var-declaration-per-line

Reply:
I think we should avoid such abbreviations, we have already two people stumbling upon this here respectively not knowing how this would behave (with me it would be three 😋) and this just makes code less approachable if 50% of developers do not know how to read the code.

(sometimes a bit difficult to draw the lines but I would say in this case it's relatively clear)

Reply:
Good point, will update! :)